user/reference/overview.md

```diff
@@ -37,6 +47,42 @@ The following table summarizes the differences between the virtual environments:
 | Memory           | 7.5 GB                                | 4 GB max                          | 7.5 GB                               | 4 GB                         |
 | Cores            | ~2, bursted                           | 2                                 | ~2, bursted                          | 2                            |
 
+## What infrastructure is my environment running on?
+
+Usually, knowing the virtualization environment characteristics from the [table above](#Virtualisation-Environment-vs-Operating-System) is sufficient.
+
+But, if you do need more detail, you have one of these two questions:
+
+* you want to see what infrastructure a [finished build](#For-a-finished-build) ran on.
+* you want to determine what infrastructure a [particular `.travis.yml` configuration](#For-a-particular-.travis.yml-configuration) will run on.
+
+### For a finished build
+
+To see what infrastructure a finished build ran on, look at the *hostname* at the top of the build log:
+
+![Infrastructure shown in hostname](/images/ui/what-infrastructure.png "Infrastructure shown in hostname")
+
+if it contains:
+
+* `ec2`, as in the previous example → the build ran in a container-based environment on Amazon EC2.
+* `gce` → the build ran in a sudo-enabled environment on Google Cloud Engine.
+* `jupiter` → the build ran on macOS
+
+### For a particular .travis.yml configuration
+
+Many different parts of your `.travis.yml` affect what infrastructure your build runs on. The following list describes some of the main settings that determine build routing:
+
+* Any of the following sudo or docker related settings route your build to a sudo-enabled linux (Ubuntu Trusty) environment on Google Cloud Engine.
+
+  - `services: docker`
+  - `sudo: required` or `sudo: true`
+  - *any* other `sudo` command in your build script
+  _ *any* other `docker` command in your build script
+
+* Using `os: osx`, setting a version of OS X using `osx_image:`, or using a macOS specific language such as `language: objective-c` routes your build to macOS infrastructure.
```

Comment:
I know it's confusing but osx_image will rather set a version of Xcode.

Reply:
Thanks @cotsog. Fixed.